presto-spi/src/main/java/com/facebook/presto/spi/ServerInfo.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi;
+
+import java.util.Objects;
+
+import static java.util.Objects.requireNonNull;
+
+public class ServerInfo
+{
+    private final String nodeId;
+    private final String environment;
+    private final String version;
+
+    public ServerInfo(String nodeId, String environment, String version)
+    {
+        this.nodeId = requireNonNull(nodeId, "nodeId is null");
+        this.environment = requireNonNull(environment, "environment is null");
+        this.version = requireNonNull(version, "version is null");
+    }
+
+    public String getNodeId()
+    {
+        return nodeId;
+    }
+
+    public String getEnvironment()
+    {
+        return environment;
+    }
+
+    public String getVersion()
+    {
+        return version;
+    }
+
+    @Override
+    public boolean equals(Object o)
```

Comment:
Do we need equals / hashCode? This doesn't seem like the type of class you'd use in a collection, or that you would be comparing. I don't have a great sense of when is appropriate to implement these, but I prefer to leave these out until it's clear how and why they are needed.